app/lib/features/chat_ng/providers/chat_room_messages_provider.dart

```diff
@@ -55,6 +79,33 @@ final renderableChatMessagesProvider =
   }).toList();
 });
 
+// Provider to check if we should show avatar by comparing with the next message
+final shouldShowAvatarProvider = Provider.family<bool, RoomMsgId>(
+  (ref, roomMsgId) {
+    final roomId = roomMsgId.$1;
+    final eventId = roomMsgId.$2;
+    final messages = ref.watch(renderableChatMessagesProvider(roomId));
+    final currentIndex = messages.indexOf(eventId);
+
+    // Always show avatar for the first message (last in the list)
+    if (currentIndex == messages.length - 1) return true;
+
+    // Get current and next message
+    final currentMsg = ref.watch(chatRoomMessageProvider(roomMsgId));
+    final nextMsg = ref.watch(
+      chatRoomMessageProvider((roomId, messages[currentIndex + 1])),
+    );
```

Comment:
I wonder if we shouldn't just make that its own provider, that would have the benefit of making caching nicer _and_ mocking easier if we just did a `nextMessageProvider(eventId) -> Message?` ...

Reply:
Yeah, good idea. This can be degraded for ease in testing 👍🏻 !